### PR TITLE
Issue/Restarting projection during event processing

### PR DESCRIPTION
**Pull request by grzegorzstachniukalm**
#176

### CODE REVIEW

File: src/Projection/PdoEventStoreProjector.php [hunk 1]
```diff
@@ -1025,4 +1028,28 @@ private function quoteTableName(string $tableName): string
...
+
+    private function startAgain(): void
+    {
+        $this->isStopped = false;
+
+        $newStatus = ProjectionStatus::RUNNING();
+
+        $projectionsTable = $this->quoteTableName($this->projectionsTable);
+        $startProjectionSql = <<<EOT
+UPDATE $projectionsTable SET status = ? WHERE name = ?;
```
Comment (prolic) [inline src/Projection/PdoEventStoreProjector.php]: LockUntil should be updated as well I think

**Comment by prolic** [comment]
ping @grzegorzstachniukalm

**Comment by grzegorzstachniukalm** [comment]
> ping @grzegorzstachniukalm

I'm on it

**Comment by grzegorzstachniukalm** [comment]
@prolic ready i think :)

File: src/Projection/PdoEventStoreReadModelProjector.php [hunk 2]
```diff
@@ -984,4 +988,35 @@ private function quoteTableName(string $tableName): string
...
+        $now = new DateTimeImmutable('now', new DateTimeZone('UTC'));
+
+        $projectionsTable = $this->quoteTableName($this->projectionsTable);
+        $startProjectionSql = <<<EOT
+UPDATE $projectionsTable SET status = ?, locked_until = ? WHERE name = ?;
+EOT;
+        $statement = $this->connection->prepare($startProjectionSql);
+        try {
+            $statement->execute([
+                                    $newStatus->getValue(),
```
Comment (prolic) [inline src/Projection/PdoEventStoreReadModelProjector.php]: a `little` too much indentation

File: tests/Projection/MySqlEventStoreReadModelProjectorTest.php [hunk 3]
```diff
@@ -172,4 +174,45 @@ function (array $state, Message $event): array {
...
+        ]);
+
+        $projectionManager = $this->projectionManager;
+
+        $eventCounter = 0;
+
+        $projection
+            ->fromAll()
+            ->when([
+                       UserCreated::class => function ($state, Message $event) use (&$eventCounter): void {
```
Comment (prolic) [inline tests/Projection/MySqlEventStoreReadModelProjectorTest.php]: again, too much indentation

**Comment by prolic** [comment]
ping @grzegorzstachniukalm

File: tests/Projection/MySqlEventStoreProjectorTest.php [hunk 4]
```diff
@@ -152,4 +154,43 @@ function (array $state, Message $event): array {
 
         $this->assertEquals(['user' => 10], \json_decode($row['position'], true));
     }
+
+    /**
+     * @test
+     */
+    public function a_running_projector_that_is_reset_should_keep_stream_positions(): void
```
Comment (prolic) [inline tests/Projection/MySqlEventStoreProjectorTest.php]: shouldn't that be removed as it is in the base-class now?

**Pull request merged**